### PR TITLE
[NG-530] refactor: update meteor button theme

### PR DESCRIPTION
**Pull request by 0x706b**

### CODE REVIEW

File: packages/themes/src/meteor/components/Button/themeOverrides.ts [hunk 1]
```diff
@@ -85,48 +58,35 @@ export const MonorailButtonOverrides: Components<Theme>['MuiButton'] = {
...
+          left: '1px',
+          zIndex: 1,
+          borderRadius: theme.spacing(1),
+          width: 'calc(100% - 2px)',
+          height: 'calc(100% - 2px)',
+          backgroundColor: 'transparent',
+          transition: theme.transitions.create(['background-color'], {
+            duration: theme.transitions.duration.standard,
+          }),
+        },
```
Comment (0x706b) [inline packages/themes/src/meteor/components/Button/themeOverrides.ts:74]: I found a new way to layer colors with the `::before` pseudo element! It works with transitions, unlike the gradient

File: packages/themes/src/meteor/components/Button/themeOverrides.ts [hunk 2]
```diff
@@ -85,48 +58,35 @@ export const MonorailButtonOverrides: Components<Theme>['MuiButton'] = {
...
         color: theme.palette[color].contrastText,
         backgroundColor: theme.palette[color].main,
         '&:hover': {
-          boxShadow: `inset 0 0 0 1px ${theme.palette[color].border.main}`,
-          background: `linear-gradient(0deg, ${theme.palette.action.hover} 0%, ${theme.palette.action.hover} 100%), ${theme.palette[color].main}`,
+          boxShadow: `inset 0 0 0 1px ${theme.palette[color].border.dark}`,
```
Comment (gsamaniego41) [inline packages/themes/src/meteor/components/Button/themeOverrides.ts:78]: Turns out MUI applies `.dark` as the background color on hover so we'll need `backgroundColor: theme.palette[color].main` here as well. I was wondering why hover and active look identical in Storybook.

File: packages/themes/src/meteor/components/Button/themeOverrides.ts (hunk 1 above)
Comment (gsamaniego41) [reply packages/themes/src/meteor/components/Button/themeOverrides.ts:74]: Nice! This is way more manageable and cleaner. ✨

**Review by gsamaniego41 — CHANGES_REQUESTED** [review]

File: packages/themes/src/meteor/components/Button/themeOverrides.ts (hunk 2 above)
Comment (0x706b) [reply packages/themes/src/meteor/components/Button/themeOverrides.ts:78]: Gotcha, thank you!

**Review by dlosorioa — APPROVED** [review]
The colors updates looks good in the Buttons variants.

Can we please update Storybook so that the variants that are not intended to use are shown as deprecated marked in a way that it is clear?

Active variants:
- Contained in Primary
- Outline in Secondary and Error
- Text in Secondary

Deprecated:
- All Success, Warning, Info variants
- Contained for Secondary and Error
- Outline and Text for Primary
- Text for Error

**Review by gsamaniego41 — APPROVED** [review]
LGTM!

**Pull request merged**